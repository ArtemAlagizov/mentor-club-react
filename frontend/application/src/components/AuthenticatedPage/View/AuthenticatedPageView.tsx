import React from "react";
import { Styled } from "../Styled";
import TopBar from "../components/TopBar/Container/TopBar";
import { IAuthenticatedPageProps } from "../Container/IAuthenticatedPageProps";
import NotificationList from "../../NotificationList/Container/NotificationList";
import translator from "../../../config/i18next";
import { translateMentorRequestStatus } from "../../../service/MentorRequestStatusHelper/MentorRequestStatusHelper";
import { Link } from "react-router-dom";

const AuthenticatedPageView: React.FC<IAuthenticatedPageProps> = ({
  children,
  notifications,
  deleteNotification,
  logout
}) => {
  return (
    <Styled.FlexWrapper>
      <Styled.FlexBox>
        <TopBar logout={logout} />
        <NotificationList
          notifications={notifications}
          deleteNotification={deleteNotification}
          notificationRenderer={(payload: any) => (
            <>
              {translator.t("notifications.mentorRequestUpdated", {
                mentorRequestId: payload.mentorRequestId,
                newStatus: translateMentorRequestStatus(payload.newStatus)
              })}
              <br />
              <Link to={`/mentor-requests/${payload.mentorRequestId}/results`}>
                {translator.t("notifications.viewMentorRequest")}
              </Link>
            </>
          )}
        >
          <Styled.Main>{children}</Styled.Main>
        </NotificationList>
      </Styled.FlexBox>
    </Styled.FlexWrapper>
  );
};

export default AuthenticatedPageView;
