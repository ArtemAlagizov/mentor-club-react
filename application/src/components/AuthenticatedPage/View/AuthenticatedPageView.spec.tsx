import { shallow } from "enzyme";
import React from "react";
import AuthenticatedPageView from "./AuthenticatedPageView";

describe("Given a AuthenticatedPageView", () => {
  describe("when rendering", () => {
    it("should have AuthenticatedPageView matching snapshot", () => {
      const authenticatedPageView = shallow(
        <AuthenticatedPageView logout={jest.fn()}>
          <div />
          <div />
        </AuthenticatedPageView>
      );

      expect(authenticatedPageView).toMatchSnapshot();
    });
  });
});
