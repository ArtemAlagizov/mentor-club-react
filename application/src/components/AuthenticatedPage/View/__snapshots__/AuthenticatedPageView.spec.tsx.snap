// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a AuthenticatedPageView when rendering should have AuthenticatedPageView matching snapshot 1`] = `
<styled.div>
  <styled.main>
    <TopBar
      logout={[MockFunction]}
    />
    <NotificationList
      notificationRenderer={[Function]}
    >
      <styled.main>
        <div />
        <div />
      </styled.main>
    </NotificationList>
  </styled.main>
</styled.div>
`;
